### PR TITLE
 SPARK-1903: Highlight searched keyword in the history search

### PR DESCRIPTION
**Pull request by johnyleebrown**
SPARK-1903

### CODE REVIEW

File: core/src/main/java/org/jivesoftware/sparkimpl/plugin/transcripts/HistoryTranscript.java [hunk 1]
```diff
@@ -258,18 +266,47 @@ public final String buildString(List<HistoryMessage> messages){
...
+					BufferedReader buf = new BufferedReader(new StringReader(str));
+					int globalPos = 1;
+					while ((line = buf.readLine()) != null){
+						Matcher matcherTime = Pattern
+								.compile("\\(\\d\\d:\\d\\d:\\d\\d\\)\\s[a-zA-Z0-9]+:\\s")
+								.matcher(line);
+						if (matcherTime.find()) {
+							for (int i = matcherTime.end(); i + text.length() < line.length() + 1; i++) {
+								String match = line.substring(i, i + text.length());
+								if (text.toLowerCase().equals(match.toLowerCase())) {
```
Comment (guusdk) [inline core/src/main/java/org/jivesoftware/sparkimpl/plugin/transcripts/HistoryTranscript.java]: This does the same, but is somewhat easier on the eyes: ``text.equalsIgnoreCase( match )``

File: core/src/main/java/org/jivesoftware/sparkimpl/plugin/transcripts/HistoryTranscript.java [hunk 2]
```diff
@@ -258,18 +266,47 @@ public final String buildString(List<HistoryMessage> messages){
...
+									window.getHighlighter().addHighlight(globalPos + i,
+											globalPos + i + text.length(), highlighter);
+								}
+							}
+						}
+						globalPos = globalPos + line.length() + 1;
+					}
+				}
+			} catch (BadLocationException | IOException ex) {
+				ex.printStackTrace();
```
Comment (guusdk) [inline core/src/main/java/org/jivesoftware/sparkimpl/plugin/transcripts/HistoryTranscript.java]: Please don't do this - always print exceptions to a Log object!

**Comment by johnyleebrown** [comment]
@guusdk Done.

**Comment by guusdk** [comment]
Okay, thanks. I see that you now do a Log as well as a printStackTrace, which is also done in other places of this class. In the future, we should not do that - we should never use printStackTrace, only use the Logging framework. PrintStackTrace will always output to stdout, while the logging framework is configurable (and will put errors to the logfiles). In short never use printStackTrace, always use log. :)

**Comment by johnyleebrown** [comment]
@guusdk Got it, thanks)

**Comment by wrooot** [comment]
It doesn't highlight text in URLs. It only highlights one instance of searched word (which is in the center of the window). I guess showing the context is messing with it (the searched word can be in the context if it was repeated a few times).
image
image

**Comment by johnyleebrown** [comment]
@wrooot 

screen shot 2017-04-23 at 11 11 57

Links
screen shot 2017-04-23 at 11 14 34

**Comment by wrooot** [comment]
I guess it depends on the amount of messages or some other stuff. Sometimes it works, sometimes don't. We can mark it fixed for now, unless someone comes up with an exact way to reproduce it.

**Pull request merged**